pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -181,6 +183,13 @@ func (c *Cluster) Start() error {
 		logger.Warningf("useAllNodes is set to false and no nodes, driveGroups, storageClassDevicesets or volumeSources are specified, no OSD pods are going to be created")
 	}
 
+	if c.spec.WaitTimeoutForHealthyOSD != 0 {
+		c.clusterInfo.OsdUpgradeTimeout = (c.spec.WaitTimeoutForHealthyOSD * 60) * time.Second
```

Comment:
Instead of setting a new `OsdUpgradeTimeout` on the clusterInfo, how about passing `c.spec` where necessary so the OSD upgrader can check this setting for itself? Then we could avoid the copied setting.

Reply:
1. The `UpdateCephDeploymentAndWait` method is used in a lot of places including the nfs operator. If `c.spec` is passed as an argument, then its only purpose would be to use the new `WaitTimeoutForHealthyOSD`. Daemons other than OSD, don't need this new field.
2. We already use 7 arguments in `UpdateCephDeploymentAndWait` 

I was able to make the changes and use `c.spec` as argument and test it. But didn't look like a better approach than using the clusterInfo. Thoughts?